### PR TITLE
python3Packages.astropy: apply patch removing the usage of polyfill.io

### PR DESCRIPTION
**Pull request by doronbehar**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/astropy/default.nix [hunk 1]
```diff
@@ -37,10 +38,13 @@ buildPythonPackage rec {
     hash = "sha256-5cb0XZEcMKy41VbH+O2ZSuxxsQjmHu5QZ/AK8eTjYTg=";
   };
 
-  postPatch = ''
-    substituteInPlace pyproject.toml \
-      --replace-fail "numpy>=2.0.0"  "numpy"
-  '';
```
Comment (LeSuisse) [inline pkgs/development/python-modules/astropy/default.nix:43]: Not sure to understand why this removal is part of this change.

Comment (doronbehar) [reply pkgs/development/python-modules/astropy/default.nix]: > Not sure to understand why this removal is part of this change.

You are right! It was a cherry-pick + merge conflict mistake.

**Comment by vcunat** [comment]
A bit late... but the test change was also intentional?  (given that the patch only changes docs if I read it right)

**Comment by doronbehar** [comment]
Aish I didn't mean to push that test change :/, it was also part of my a bit careless `git cherry-pick`. The original author of this commit stated that the test was flaky, like a few more others, but I wanted to test that out here. In general the flaky tests might not be flaky anymore, and we might be able to remove more of them, but that will take time, and can be done in a PR against `master` if someone really cares.

**Comment by LeSuisse** [comment]
Yup I confirm I had trouble with it when I tested on master. It might be because I was building on a quite loaded machine and the test seems to be timing sensitive.

Also, @doronbehar do you want to handle the cherry-pick to stable or should I plan to do it?

**Comment by doronbehar** [comment]
> Yup I confirm I had trouble with it when I tested on master. It might be because I was building on a quite loaded machine and the test seems to be timing sensitive.
> 
> Also, @doronbehar do you want to handle the cherry-pick to stable or should I plan to do it?

Yes please go ahead.

**Pull request merged**